network/utils.py

```diff
@@ -68,8 +68,10 @@ def variational_free_energy_loss(model, scale_factor, kl_alpha):
 
     # KL Divergence should be applied once per epoch only, so
     # scale_factor should be num_samples / batch_size.
-    kl = sum(model.losses) / scale_factor
-
+    print(type(scale_factor), type(model), type(model.losses[0]))
+    # TODO Fix here sum(model.losses) / scale_factor
+    kl = scale_factor
```

Comment:
Był problem  z funkcją straty, a mianowicie z tym, że próbowała zrobić sumę po `model.losses`.

Nie wiem dlaczego to powoduje problem. Myślę, że ten [artykuł](https://www.tensorflow.org/guide/intro_to_graphs) może być pomocny w zrozumieniu tego.

Na razie żeby naprawić ten błąd próbowałem:
1. Dodać dekoratory `tf.function` nad funkcją która liczy `loss`, nad funkcją-fabryką która tworzy funkcja fabryką która tworzy `loss`,  oraz nad nimi obiema.  (i.e. różne kombinacje tego co na obrazku)

![obraz](https://user-images.githubusercontent.com/21079319/134804737-91b22610-4163-4301-a32c-e35d4598141a.png)


2. Wyłączyć `eager.execuction` (ponieważ poprzedni bład pojawiał się w pliku `..../eager.py`
![obraz](https://user-images.githubusercontent.com/21079319/134804764-724d05c7-9113-4b20-88d8-edcac8123ed4.png)